src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
> If not, it seems that GC would get stuck for the following case (with the above unwind routine):
>
> ```
> push STK_ARG_2
> << GC >>
> push STK_ARG _1
> call FUNC
> ```

The x86 gc and unwind info keeps track of how much was pushed onto stack and whether it needs protecting. E.g. look for GetPushedArgSize in eetwain.cpp - it returns how much is pushed on the stack at given offset in the method.